README.md

```diff
@@ -1,2 +1,421 @@
 # MangoJuice
 
+> The missing `M.C` part for your favorite `V` library.
+
+* **Highly scalable** because of strongly decoupled M, V and C and good definition of logical Block of your application
+* **Forces you to write good decomposed code**, which is easy to scale, test and debug.
+* **Fully-featured core**, no need to add any middlewares to do just regular things
+* **View library agnostic**
+* **Support for server-rendering**, but depends on your V lib, of course
+* **Lightweight** (zero dependencies, around **6kb minified gzipped**)
+* **Easy to learn and fun to use**
+
+## Installation
+```bash
+npm install mangojuice-core  # general set of tools
+npm install mangojuice-react # for using react for view rendering
+npm install babel-plugin-transform-decorators-legacy
+```
+MangoJuice relies on decorators, so you should add decorators transformer to your babel configuration.
+
+
+## Quick start (block, cmd, async tasks, run)
+MangoJuice (MJS) consists of three main parts:
+* **Model** – plain function `const createModel = () => ({ ... })`, which defines initial model state of your logical block.
+* **Logic** - plain object `const Logic = { ... };` with a set of commands inside.
+* **View** – plain function `const View = ({ model }) => { ... }`, which should display something based on current state of a model. If your favorite V library is React, then View is a plain-function component.
+
+All these three parts together called `Block`. For example, let's say you have a page with very complicated search form and results list. In this case you will probably have three logical blocks: (1) Search form, (2) Results list and (3) Block that manage search form and results (parent block of 1 and 2).
+
+#### Search Form example
+```js
+// SearchForm.js
+import { Cmd } from 'mangojuice-core';
+
+export const createModel = () => ({
+  query: ''
+});
+export const Logic = {
+  name: 'SearchForm',
+
+  @Cmd.update
+  SetQuery(ctx, e) {
+    return { query: e.target.value };
+  },
+
+  @Cmd.nope
+  Search() {
+  }
+};
+export const View = ({ model, exec }) => (
+  <div>
+    <h2>Complicated Search Form</h2>
+    <input value={model.query} onChange={exec(Logic.SetQuery)} /><br />
```

Comment:
Ok, i found a way without using a transformer :) I just teach react to work with commands. But i'm not sure about performance, i'm replacing `React.createElement` with my own wrapper, which is checking each prop to find a command, and if some prop is a command – replace it with function that will exec command in current model context.